synapse/types.py

```diff
@@ -751,3 +751,32 @@ def get_verify_key_from_cross_signing_key(key_info):
     # and return that one key
     for key_id, key_data in keys.items():
         return (key_id, decode_verify_key_bytes(key_id, decode_base64(key_data)))
+
+
+@attr.s(frozen=True, slots=True)
+class User:
+    """Represents a user object.
+
+    Attributes:
+        user:  ID of the user.
+        admin:  True if the user is an admin.
+        app_service:  Application service that created this user.
+        consent_server_notice_sent:  Version of policy documents the user has been sent.
+        consent_version:  Version of policy documents the user has consented to.
+        creation_ts:  Creation timestamp of the user.
+        deactivated:  True if the user has been deactivated.
+        is_guest:  True if the user is a guest user.
+        shadow_banned:  True if the user has been shadow-banned.
+        user_type:  User type (None for normal user, 'support' and 'bot' other options).
+    """
+
+    user = attr.ib(type="UserID")
```

Comment:
nowadays this can be written:

```
user: UserID = attr.ib()
```

... or, even more succinctly, along with `attr.s(auto_attribs=True)`:

```
user: UserID
```

(see https://www.attrs.org/en/stable/types.html?highlight=auto_attribs#type-annotations)


If you're using auto_attribs, default values can also be given this way:

```
admin: bool = False
```

- not that I think you actually need defaults, since all the values are set explicitly.

I won't insist on you changing these, but I thought it might be useful to know for the future at least.